quack/quack.py

```diff
@@ -113,22 +115,27 @@ def _fetch_modules(config, specific_module=None):
             subprocess.call('rm .gitmodules'.split())
             subprocess.call('git rm --quiet --cached .gitmodules'.split())
 
-        print('Cloned: ' + module[0] + (tag or hexsha))
+        print('Cloned: ' + folder_name + (tag or hexsha))
 
         if config.get('gitignore'):
             with open('.gitignore', 'a') as file_pointer:
-                if module[0] not in ignore_list:
-                    file_pointer.write('\n' + module[0])
-                    ignore_list.append(module[0])
+                if folder_name not in ignore_list:
+                    file_pointer.write('\n' + folder_name)
+                    ignore_list.append(folder_name)
 
 
 def _clean_modules(config, specific_module=None):
     """Remove all given modules."""
     for module in config.get('modules').items():
         if specific_module and specific_module != module[0]:
             continue
-        if _remove_dir(module[0]):
-            print('Cleaned', module[0])
+
+        folder_name = module[1].get('foldername')
```

Comment:
If `foldername` not available, then it should take module name i.e. `module[0]` as `foldername` and clean that folder.